src/node-lib/educator-api/helpers/saveUnits/useSaveUnits.tsx

```diff
@@ -0,0 +1,66 @@
+import { OakP } from "@oaknational/oak-components";
+import { useState, useCallback } from "react";
+import { useUser } from "@clerk/nextjs";
+
+import { useOakToastContext } from "@/context/OakToast/useOakToastContext";
+import { postEducatorData } from "@/node-lib/educator-api/helpers/postEducatorData";
+
+export const useSaveUnits = (
+  savedUnits: Array<string>,
+  programmeSlug: string,
+) => {
+  const [locallySavedUnits, setLocallySavedUnits] = useState<string[]>([]);
+  const { isSignedIn } = useUser();
+
+  const isUnitSaved = useCallback(
+    (unitSlug: string) =>
+      savedUnits?.includes(unitSlug) || locallySavedUnits.includes(unitSlug),
+    [savedUnits, locallySavedUnits],
+  );
+
+  const { setCurrentToastProps } = useOakToastContext();
+
+  const onSave = async (unitSlug: string) => {
+    setLocallySavedUnits((prev) => [...prev, unitSlug]);
+    setCurrentToastProps({
+      message: (
+        <OakP>
+          <b>Unit saved</b> to My Library
```

Comment:
Just making a note here that I think this should eventually display a more specific message eg. "Unit saved to English KS1". I'll add it to the AC if it's not already included.